### DISCUSSION TITLE
Is this library compatible with apollo caching behaviour ?

### DISCUSSION THREAD

**Discussion by ajacquierbret** [badoo/MVIKotlin#223] [Q&A]
@arkivanov  I'm very interested in the MVI pattern and Kotlin Multiplatform, and this library seems like a perfect fit for my project.
But being used to apollo-graphql ([apollo-android](https://github.com/apollographql/apollo-android)) and its caching features, I'm curious to know if it'd be possible to use it instead of the MVIKotlin's store interface. Do you think it's feasable ? Does anyone heard of something like that ?

**Comment by arkivanov** [comment]
The `Store` interface is the core component of the library, everything is built around it. How do you imagine using the library and at the same time using something instead of the `Store` interface?

**Comment by arkivanov** [comment]
Converted the issue to the discussion.

**Reply by ajacquierbret** [reply]
Thanks for your very fast answer. I was just thinking about a way to implement the MVI pattern on something like a graphql cache, and especially the one built by the Apollo team. But I understand that it isn't possible with the MVIKotlin library, and even maybe with every other library providing an abstraction layer above this pattern, because the state is the core feature of it. Anyway, thanks again !

**Reply by arkivanov** [reply]
I'm not really familiar with the graphql caching, but you may think about implementing the `Store` and the `StoreFactory` interfaces in your way. If this would work for you, you will still be using the library and will get its features like the `Binder`, time traveling, logging, etc.

**Reply by ajacquierbret** [reply]
It would be great ! I'll dig a bit more into both sides and maybe try to build my own implementation of the store. I'll ping you if I finally achieve to do it. Thanks again.
